skops/hub_utils/_hf_hub.py

```diff
@@ -205,14 +238,69 @@ def recursively_default_dict() -> MutableMapping:
         config["sklearn"]["example_input"] = _get_example_input(data)
         config["sklearn"]["columns"] = _get_column_names(data)
     elif "text" in task:
-        if isinstance(data, list) and all(isinstance(x, str) for x in data):
-            config["sklearn"]["example_input"] = {"data": data[:3]}
-        else:
-            raise ValueError("The data needs to be a list of strings.")
+        error_message = "The data needs to be an iterable of strings."
+        try:
+            data_head = _head(data, n=3)
+            if _is_sequence_of_strings(data_head):
+                config["sklearn"]["example_input"] = {"data": data_head}
+            else:
+                raise ValueError(error_message)
+        except TypeError as e:
+            raise ValueError(error_message) from e
 
     dump_json(Path(dst) / "config.json", config)
 
 
+def _head(data: Iterable, n: int):
+    """Returns the first n elements of data.
+
+    Raises a ``TypeError`` if data is not an iterable.
+
+    Parameters
+    ----------
+    data: Iterable
+        Any iterable.
+
+    n: int
+        Number of elements to extract from the head of data.
+
+    Raises
+    ------
+    TypeError
+        If data is not an iterable (raised by itertools.islice).
+
+    Returns
+    -------
+    data_head: list
+        A list containing the first n elements of data.
+    """
+    data, data_copy = itertools.tee(data, 2)
+    return list(itertools.islice(data_copy, n))
+
+
+def _is_sequence_of_strings(data):
```

Comment:
to me it seems better to remove this method and have it in the function where it's called.